impeller/aiks/experimental_canvas.cc

```diff
@@ -189,6 +222,13 @@ void ExperimentalCanvas::SaveLayer(
 
   auto result = inline_pass_contexts_.back()->GetRenderPass(0u);
   render_passes_.push_back(result.pass);
+
+  // Start non-collapsed subpasses with a fresh clip coverage stack limited by
+  // the subpass coverage. This is important because image filters applied to
+  // save layers may transform the subpass texture after it's rendered,
+  // causing parent clip coverage to get misaligned with the actual area that
+  // the subpass will affect in the parent pass.
+  clip_coverage_stack_.PushSubpass(subpass_coverage, GetClipHeight());
```

Comment:
Ah, right. This applies when I am deducing a cull rect from the existing clip area, not when the bounds are supplied. DlBuilder has always used the supplied bounds for culling. In your case the bounds should always be supplied. It might apply if the code at the top of the function that uses the render target size as the layer bounds is ever invoked, but that should be vestigial at this point.

Reply:
So, to be more specific, if there is an existing cull rect, it should not be used to cull the contents of a saveLayer with a filter applied - even if that existing cull rect is "the bounds of the parent drawable/layer". You can either reverse transform the drawable bounds through the filter (not through the transform) or use an infinite cull rect.

The bounds from the saveLayer, though, can be applied.

Consider that this isn't just the case of expanding or contracting the bounds for a blur radius - another ImageFilter transforms the pixels by a coordinate transform and so the pixels that will end up on the parent drawable could be rendered with any coordinates, even ones that bear no resemblance to the destination drawable. So, do not inherit any "culling rect" without consulting the filter.

In this case, the DlBuilder is already promising bounds, so that case is legacy/not-a-priority. I would just replace the "if we didn't get a bounds" code at the top of the function with "assert(we got bounds)" and this whole issue goes away.

Reply:
There is still a saveLayer without bounds here:

https://github.com/flutter/engine/blob/main/impeller/display_list/dl_dispatcher.cc#L1028C17-L1030

Reply:
It has a ToRect in there...? And it claims "kContainsContents"...? Doesn't that mean "I have bounds and you can trust them?